File: Dot examples/ASP Book Examples/Book Examples/Chapter15/CS/App_Code/PropertyHelloWorld.cs
using System;

public class PropertyHelloWorld
{

    private string _message;

    public string Message
    {
        get
        {
            return _message;
        }
        set
        {
            if (value.Length > 5)
                throw new Exception("Message too long!");
            _message = value;
        }
    }

    public string SayMessage()
    {
        return _message;
    }

}

File: Dot examples/ASP Book Examples/Book Examples/Chapter15/VB/App_Code/PropertyHelloWorld.vb
Imports System

Public Class PropertyHelloWorld

    Private _message As String

    Public Property Message() As String
        Get
            Return _message
        End Get
        Set(ByVal Value As String)
            If Value.Length > 5 Then
                Throw New Exception("Message too long!")
            End If
            _message = Value
        End Set
    End Property

    Public Function SayMessage() As String
        Return _message
    End Function

End Class
